### PR TITLE
Updated for CakePHP 4.x

### PR DESCRIPTION
**Pull request by mirko-pagliai**
We need to create a new branch where to merge

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -30,11 +30,12 @@
...
   "require": {
-    "php": ">=5.6",
-    "cakephp/cakephp": "^3.5"
+    "php": ">=7.2.0",
+    "cakephp/cakephp": "4.x-dev",
+    "cakephp/chronos": "2.x-dev as 2.0.0"
   },
   "require-dev": {
-    "phpunit/phpunit": "^5.7.14|^6.0",
+    "phpunit/phpunit": "^7.0|^8.0",
```
Comment (ADmad) [inline composer.json]: This should be `^8.0`.

File: tests/bootstrap.php [hunk 2]
```diff
@@ -23,4 +21,4 @@
 chdir($root);
 require $root . '/vendor/cakephp/cakephp/tests/bootstrap.php';
 
-Plugin::load('Recaptcha', ['path' => dirname(dirname(__FILE__)) . DS]);
+//Plugin::load('Recaptcha', ['path' => dirname(dirname(__FILE__)) . DS]);
```
Comment (ADmad) [inline tests/bootstrap.php]: Just remove line instead of commenting

File: tests/TestCase/View/Helper/RecaptchaHelperTest.php [hunk 3]
```diff
@@ -11,10 +11,12 @@
 class RecaptchaHelperTest extends TestCase
 {
 
-    public function setUp()
+    public function setUp(): void
     {
         parent::setUp();
 
+        $this->loadPlugins(['Recaptcha']);
```
Comment (ADmad) [inline tests/TestCase/View/Helper/RecaptchaHelperTest.php:18]: I don't think this is necessary. Just the autoloading provided by composer should be enough.

Comment (mirko-pagliai) [reply tests/TestCase/View/Helper/RecaptchaHelperTest.php:18]: Without it, it does not find the `templates/element/recaptcha.php` element

Comment (ADmad) [reply tests/TestCase/View/Helper/RecaptchaHelperTest.php:18]: Ah, I overlooked that we need to use templates too which are not autoloaded :slightly_smiling_face:.

**Comment by ADmad** [comment]
I have targeted the PR to new `cake-4` branch. We can merge if you remove draft status.

**Comment by mirko-pagliai** [comment]
> I have targeted the PR to new `cake-4` branch. We can merge if you remove draft status.

Done. Thanks.

**Comment by mirko-pagliai** [comment]
@ADmad how can we do to have the same branch also in the `agiletechvn/Recaptcha` repository, so that it is also visible for packagist?

**Comment by ADmad** [comment]
@mirko-pagliai I don't have commit rights to that repo so I can't do anything about that. On packagist the plugin is under the name [crabstudio/recaptcha](https://packagist.org/packages/crabstudio/recaptcha). So ideally that should be updated to use this new repo. Only @anhtuank7c can do that.

**Comment by ADmad** [comment]
For the time being if you want to use the `cake-4` branch from this repo you can do so using composer's `repositories` config.

**Comment by mirko-pagliai** [comment]
> So ideally that should be updated to use this new repo. Only @anhtuank7c can do that.
I hope this, it seems to me the most sensible thing, even to keep the code updated

I hope this, it seems to me the most sensible thing, even to keep it updated

> For the time being if you want to use the `cake-4` branch from this repo you can do so using composer's `repositories` config.

Yes, I know. But it would be better to change repository, so that someone could seem to keep the code up to date with CakePHP

@anhtuank7c please, tell us

**Pull request merged**